### ISSUE TITLE
seems like hot reloading doesn't work with react-router 1.0.0?

### ISSUE THREAD

**Issue by jdeal** [gaearon/react-hot-loader#212]
react-hot-loader works fine with a normal component tree, but as soon as I introduce react-router (1.0.0), it stops working for children of the router.

My index is:

```jsx
render(
  <Root/>,
  document.getElementById('root')
);
```

If my `Root` component is:

```js
export default class Root extends Component {
  render() {
    return <HomePage/>;
  }
}
```

then changes to `HomePage` are hot reloaded just fine. But if `Root` is:

```jsx
export default class Root extends Component {

  render() {
    return (
      <Router>
        <Route path="/" component={App}>
          <Route path="home" component={HomePage}/>
        </Route>
      </Router>
    );
  }
}
```

then changes to `HomePage` are not hot reloaded. No errors on the console. It reports that it's reloaded:

```
[WDS] App updated. Recompiling...
client?e636:78 [WDS] App hot update...
only-dev-server.js:69 [HMR] Checking for updates on the server...
log-apply-result.js:20 [HMR] Updated modules:
log-apply-result.js:22 [HMR]  - 302
log-apply-result.js:22 [HMR]  - 308
only-dev-server.js:55 [HMR] App is up to date.
```

If I flip to another route and back (via links), that gets it to render the changes. (Or of course if I refresh.)

Do I have to do anything special for it to hot reload? This worked fine with react-router 0.13.3 and `Router.run`.

Thanks!

**Comment by jdeal** [comment]
After sleeping on it, I figured out the problem. My `HomePage` component in the above example was a stateless function component like:

```jsx
const HomePage = () =>
  <div>
    <Greeting/>
  </div>;
```

If instead I make it:

```jsx
class HomePage extends React.Component {
  render() {
    return (
      <div>
        <Greeting/>
      </div>
    );
  }
}
```

then hot reloading works fine. Nested child components (like `<Greeting>` in that example) can be stateless and also work fine. But it seems that components passed to `<Route>` must be normal components. This seems like a good thing to document somewhere, but I'm not sure if this is a requirement that's specific to react-router or somehow general to react-hot-loader.

Closing this issue since I found a solution.
